includes/data-port/models/class-sensei-import-lesson-model.php

```diff
@@ -0,0 +1,496 @@
+<?php
+/**
+ * File containing the Sensei_Import_Lesson_Model class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Handles the import for lessons and quizzes.
+ */
+class Sensei_Import_Lesson_Model extends Sensei_Import_Model {
+
+	/**
+	 * Lesson's course.
+	 *
+	 * @var int
+	 */
+	private $course_id;
+
+	/**
+	 * Create a new lesson or update an existing lesson.
+	 *
+	 * @return true|WP_Error
+	 */
+	public function sync_post() {
+		$result = $this->sync_lesson();
+
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+
+		$result = $this->sync_quiz();
+
+		return is_wp_error( $result ) ? $result : true;
+	}
+
+	/**
+	 * Helper method which stores the quiz post.
+	 *
+	 * @return bool|WP_Error  True on success, WP_Error on failure.
+	 */
+	private function sync_quiz() {
+		$quiz_args = $this->get_quiz_args();
+
+		if ( is_wp_error( $quiz_args ) ) {
+			return $quiz_args;
+		}
+
+		$quiz_id = wp_insert_post( $quiz_args, true );
+
+		if ( is_wp_error( $quiz_id ) ) {
+			return $quiz_id;
+		}
+
+		if ( $this->is_new() ) {
+			wp_set_post_terms( $quiz_id, array( 'multiple-choice' ), 'quiz-type' );
+		}
+
+		update_post_meta( $this->get_post_id(), '_lesson_quiz', $quiz_id );
+
+		$result = $this->set_quiz_questions( $quiz_id );
+
+		return is_wp_error( $result ) ? $result : true;
+	}
+
+	/**
+	 * Helper method to get quiz post arguments.
+	 *
+	 * @return array  The arguments.
+	 */
+	private function get_quiz_args() {
+		$args = [
+			'post_type'    => 'quiz',
+			'post_parent'  => $this->get_post_id(),
+			'post_content' => '',
+		];
+
+		if ( $this->is_new() ) {
+			$current_user        = get_current_user_id();
+			$args['post_author'] = $current_user ? $current_user : $this->get_default_author();
+		} else {
+			$args['ID'] = Sensei()->lesson->lesson_quizzes( $this->get_post_id() );
+		}
+
+		$value = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_STATUS );
+		if ( null !== $value ) {
+			$args['post_status'] = $value;
+		}
+
+		$value = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_TITLE );
+		if ( null !== $value ) {
+			$args['post_title'] = $value;
+		}
+
+		$meta = $this->get_quiz_meta();
+
+		if ( is_wp_error( $meta ) ) {
+			return $meta;
+		}
+
+		$args['meta_input'] = $meta;
+
+		return $args;
+	}
+
+	/**
+	 * Helper method to get quiz meta arguments.
+	 *
+	 * @return array|WP_Error  The arguments on success, WP_Error on failure.
+	 */
+	private function get_quiz_meta() {
+		$meta = [ '_quiz_lesson' => $this->get_post_id() ];
+
+		$pass_required = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_PASS_REQUIRED );
+		$pass_mark     = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_PASSMARK );
+
+		if (
+			( empty( $pass_required ) && ! empty( $pass_mark ) ) ||
+			( ! empty( $pass_required ) && empty( $pass_mark ) ) ) {
+			return new WP_Error(
```

Comment:
PHPCS: bump last `) {` to new line.

Reply:
https://github.com/Automattic/sensei/commit/ac19589b8a62ed55cb36a6fc740e85f929d87e28